src/app/routes/storage.ts

```diff
@@ -323,6 +332,39 @@ export class StorageController {
       });
   }
 
+  public async moveItemsToTrash(req: Request, res: Response): Promise<void> {
+    const { behalfUser: user } = req as SharedRequest;
+    const { items } = req.body;
+    
+
+    items.forEach(async (item: {id: string, type: string}) => {
```

Comment:
Consider using `for of` clause instead of forEach as forEach will not wait for anything. 

I do not recommend using `Promise.all + map` as that kind of parallelism is not controlled.

For instance, try this: 
```
async function foo() {
  const collection = [1,2,3,4,5];

  collection.forEach(async (el) => {
    await new Promise(r => setTimeout(r, 1000));
    console.log('resolved');
  });

  // something that should be run after the collection.forEach()
  console.log('finish');
}

foo();
```

The flow is not what you will expect.